generics/middleware/checkAdminRole.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * name : checkAdminRole.js
+ * author : Mallanagouda R Biradar
+ * Date : 7-Aug-2025
+ * Description : checkAdminRole middleware.
+ */
+
+const jwt = require('jsonwebtoken')
+
+module.exports = async function (req, res, next) {
+	// Define paths that require admin role validation
+	let adminPath = ['admin/deleteResource']
+	// Initialize response object for error formatting
+	let rspObj = {}
+	// Flag to check if the current request path needs admin validation
+	let checkAdminRole = false
+
+	// Check if the incoming request path matches any admin paths
+	await Promise.all(
+		adminPath.map(async function (path) {
+			if (req.path.includes(path)) {
+				checkAdminRole = true
+			}
+		})
+	)
+
+	// If path needs admin check, validate the user's role using JWT token
+	if (checkAdminRole) {
+		// Get token from request headers
+		token = req.headers['x-auth-token']
+		// If no token found, return unauthorized error
+		if (!token) {
+			rspObj.errCode = CONSTANTS.apiResponses.TOKEN_MISSING_CODE
+			rspObj.errMsg = CONSTANTS.apiResponses.TOKEN_MISSING_MESSAGE
+			rspObj.responseCode = HTTP_STATUS_CODE['unauthorized'].status
+			return res.status(HTTP_STATUS_CODE['unauthorized'].status).send(respUtil(rspObj))
+		}
+
+		let decodedToken
+		try {
```

Comment:
@MallanagoudaB I think there is option to use keycloak as well. in main middleware function you can see the logic for that as well.